lib/view.js

```diff
@@ -0,0 +1,230 @@
+/** @babel */
+
+import {Emitter} from 'event-kit';
+
+import {Path} from './models';
+import {cachedProperty, closest, dom} from './utils';
+
+
+export default class AdvancedOpenFileView {
+    constructor() {
+        this.emitter = new Emitter();
+        this.cursorIndex = null;
+        this._updatingPath = false;
+
+        // Element references
+        this.pathInput = this.content.querySelector('.path-input');
+        this.pathList = this.content.querySelector('.list-group');
+        this.parentDirectoryListItem = this.content.querySelector('.parent-directory');
```

Comment:
Huh? Did you mean to leave a comment on the `pathEditor` stuff below? I'm going to assume you did because references to DOM elements seems to be absolutely what the view should care about. Or am I mistaken?

As for `pathEditor`, while I am interacting the the model object behind the editor, it's only because that's the only way to make changes to the UI element. I guess I could store a reference to the model in the controller and have the view file listing triggered to update automagically when it changes, instead of having the controller call `setPathList`. I'd consider it for a future PR rather than bother changing this one, though, as it's just a technical design improvement and doesn't actually affect our behavior at all.